home.asm

```diff
@@ -964,7 +965,11 @@ CheckTrainerBattle:: ; 360d
 	pop de
 	pop bc
 	and a
-	jr z, .startbattle
+	jr nz, .next
+	
+; And is unoccluded
+	farcall CheckOccludingObjects ; rst farcall says it retains "a, bc, de, and hl" but return farcall says it tries to save f?
```

Comment:
That comment is referring to the register values of the caller. Commit 90a7bfbbee7303b681cdcaaca0ccf402013d7575 stopped preserving the caller's flags, but still preserves the flags when returning.